src/snowflake/connector/auth/by_plugin.py

```diff
@@ -89,18 +59,37 @@ class AuthType(Enum):
 class AuthByPlugin(ABC):
     """External Authenticator interface."""
 
-    def __init__(self) -> None:
-        self._retry_ctx = AuthRetryCtx()
+    def __init__(
+        self,
+        timeout: int | None = None,
+        max_retry_attempts: int | None = None,
+        socket_timeout_override: int | None = None,
+        backoff_policy: BackoffPolicy | None = None,
+    ) -> None:
         self.consent_cache_id_token = False
-        self._timeout: int = 120
+
+        if timeout is None:
+            timeout = DEFAULT_AUTH_CLASS_TIMEOUT
```

Comment:
Then we could just default to this instead of `None`

Reply:
The problem is that the argument might be `None`, so if we default to `DEFAULT_AUTH_CLASS_TIMEOUT` then `timeout = None` will overwrite it.